website/docs/feature-flag-tutorials/react/examples.md

```diff
@@ -0,0 +1,627 @@
+
+We make it easy to get our data, your application, and an analytics tool connected so you can collect, analyze, and report relevant data for your teams.
+
+Let’s walk through how to enable impression data for the feature flag we created from the React tutorial and capture the data in our app for analytics usage.
+
+
+### Enable feature flag impression data
+
+At the flag level in Unleash, navigate to the [Settings view](http://localhost:4242/projects/default/features/newNotificationsBadge/settings).
+
+
+![Edit Settings image](/img/react-ex-edit-settings.png)
+
+
+In the Settings view, click on the edit button. This will take us to the ‘Edit Feature toggle’ form.
+
+
+![Enable impression data image](/img/react-ex-enable-impression-data.png)
+
+
+Turn on the impression data and hit ‘Save’. Events will now be emitted every time the feature flag is triggered.
+
+You can also use our API command to enable the impression data:
+
+```
+curl --location --request PATCH 'http://localhost:4242/api/admin/projects/default/features/newNotificationsBadge' \
+    --header 'Authorization: INSERT_API_KEY' \
+    --header 'Content-Type: application/json' \
+    --data-raw '[
+  {
+    "op": "replace",
+    "path": "/impressionData",
+    "value": true
+  }
+]'
+```
+
+### Capture impression data for flag analytics
+
+Next, let’s configure our React app to capture the impression events that are emitted when our flag is triggered. To do this, we can import the [useUnleashClient hook from the React SDK](https://github.com/Unleash/proxy-client-react/blob/main/src/useUnleashClient.ts) into our app. This hook allows us to interact directly with the Unleash client to listen for events and log them.
+
+In `NavBar.tsx`, pull in `useUnleashClient`. Our updated import line should look like this:
+
+```js
+import { useFlag, useVariant, useUnleashClient } from "@unleash/proxy-client-react";
+```
+
+Next, within the `NavBar` component itself, call `useUnleashClient` and wrap a `useEffect` hook around our function calls to grab impression data with this code snippet:
+
+```js
+ const unleashClient = useUnleashClient();
+
+ useEffect(() => {
+   unleashClient.start();
+
+   unleashClient.on("ready", () => {
+     const enabledImpression = unleashClient.isEnabled("newNotificationsBadge");
+     console.log(enabledImpression);
+   });
+
+   unleashClient.on("impression", (impressionEvent: object) => {
+     console.log(impressionEvent);
+     // Capture the event here and pass it internal data lake or analytics provider
+   });
+ }, [unleashClient]);
+```
+
+This code snippet starts the Unleash client, checks that our flag is enabled, and then stores impression events for your use.
+
+> **Note:** We are passing in unleashClient into the dependency array in useEffect to prevent the app from [unnecessarily mounting the component](https://react.dev/reference/react/useEffect#parameters) if the state of the data it holds has not changed.
```

Comment:
I'd suggest wrapping code terms / variable names / etc in backticks when in plain text:

```suggestion
> **Note:** We are passing in `unleashClient` into the dependency array in `useEffect` to prevent the app from [unnecessarily mounting the component](https://react.dev/reference/react/useEffect#parameters) if the state of the data it holds has not changed.
```